interface/ceed.c

```diff
@@ -751,6 +784,66 @@ int CeedReference(Ceed ceed) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Get a `CeedVector` for scratch work from a `Ceed` context.
+
+  Note: This vector must be restored with @ref CeedRestoreWorkVector().
+
+  @param[in]  ceed `Ceed` context
+  @param[in]  len  Minimum length of work vector
+  @param[out] vec  Address of the variable where `CeedVector` will be stored
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Backend
+**/
+int CeedGetWorkVector(Ceed ceed, CeedSize len, CeedVector *vec) {
+  CeedSize i = 0;
+
+  if (!ceed->work_vectors) CeedCall(CeedWorkVectorsCreate(ceed));
+  // Search for big enough work vector
+  for (i = 0; i < ceed->work_vectors->num_vecs; i++) {
+    if (!ceed->work_vectors->is_in_use[i]) {
+      CeedSize work_len;
+
+      CeedCall(CeedVectorGetLength(ceed->work_vectors->vecs[i], &work_len));
+      if (work_len >= len) break;
+    }
+  }
+
+  // Long enough vector was not found
+  if (i == ceed->work_vectors->num_vecs) {
+    if (ceed->work_vectors->num_vecs == 0) {
+      ceed->work_vectors->num_vecs = 1;
+      CeedCall(CeedCalloc(ceed->work_vectors->num_vecs, &ceed->work_vectors->vecs));
+      CeedCall(CeedCalloc(ceed->work_vectors->num_vecs, &ceed->work_vectors->is_in_use));
+    } else {
+      ceed->work_vectors->num_vecs++;
+      CeedCall(CeedRealloc(ceed->work_vectors->num_vecs, &ceed->work_vectors->vecs));
+      CeedCall(CeedRealloc(ceed->work_vectors->num_vecs, &ceed->work_vectors->is_in_use));
```

Comment:
Formally this is a quadratic algorithm. The numbers are small so it may be harmless, but growable data structures usually decouple capacity from current size so that, e.g., capacity can be doubled each time reallocation is needed.